sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -645,6 +633,21 @@ private[sql] object DataSourceV2Strategy extends Logging {
       logWarning(log"Can't translate ${MDC(EXPR, other)} to source filter, unsupported expression")
       None
   }
+
+  private def withProjectAndFilter(
```

Comment:
Please write a function description like https://github.com/apache/spark/pull/47606/files#diff-7aeb491d44e183c8c8cf86d90b57701dba009fc19983c2a5c09449c768b047ceR549-R557 ?

Reply:
Should we change it to `protected[sql]` like the other utility function in DataSourceV2Strategy?

Reply:
Makes sense, I addressed both comments